pkg/types/v1/config.go

```diff
@@ -122,12 +122,13 @@ func (c *Config) Sanitize() error {
 }
 
 type RunConfig struct {
-	Debug          bool     `yaml:"strict,omitempty" mapstructure:"debug"`
-	Strict         bool     `yaml:"strict,omitempty" mapstructure:"strict"`
-	Reboot         bool     `yaml:"reboot,omitempty" mapstructure:"reboot"`
-	PowerOff       bool     `yaml:"poweroff,omitempty" mapstructure:"poweroff"`
-	CloudInitPaths []string `yaml:"cloud-init-paths,omitempty" mapstructure:"cloud-init-paths"`
-	EjectCD        bool     `yaml:"eject-cd,omitempty" mapstructure:"eject-cd"`
+	Debug           bool     `yaml:"debug,omitempty" mapstructure:"debug"`
+	Strict          bool     `yaml:"strict,omitempty" mapstructure:"strict"`
+	Reboot          bool     `yaml:"reboot,omitempty" mapstructure:"reboot"`
+	PowerOff        bool     `yaml:"poweroff,omitempty" mapstructure:"poweroff"`
+	CloudInitPaths  []string `yaml:"cloud-init-paths,omitempty" mapstructure:"cloud-init-paths"`
+	EjectCD         bool     `yaml:"eject-cd,omitempty" mapstructure:"eject-cd"`
+	FullCloudConfig string   // Stores the full cloud config used to generate the spec afterwards
```

Comment:
This is for ease of use. We can reuse the cloudconfig embedded here to generate the spec afterwards without having to pass it again. And maybe it will be helpful down the line to have it there?

Reply:
yes, I think so but could not make it work without a LOT of changes. Next step is merging more things together once we got 1 source of config and start de-duplicating a lot of things.

This all needs a lot of care.